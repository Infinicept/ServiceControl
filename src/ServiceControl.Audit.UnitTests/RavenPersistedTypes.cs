namespace ServiceControl.UnitTests
{
    using System;
    using System.Linq;
    using Audit.Infrastructure;
    using NUnit.Framework;
    using Particular.Approvals;
    using Raven.Client.Indexes;

    class RavenPersistedTypes
    {
        [Test]
        public void Verify()
        {
            var documentTypes = typeof(Bootstrapper).Assembly.GetTypes()
                .Where(type => typeof(AbstractIndexCreationTask).IsAssignableFrom(type))
                .SelectMany(indexType => indexType.BaseType?.GenericTypeArguments)
                .Distinct();

            var documentTypeNames = string.Join(Environment.NewLine, documentTypes.Select(t => t.AssemblyQualifiedName));

            Approver.Verify(documentTypeNames);
        }
    }
}