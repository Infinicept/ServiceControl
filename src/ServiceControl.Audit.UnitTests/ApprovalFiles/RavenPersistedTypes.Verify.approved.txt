ServiceControl.Audit.Monitoring.KnownEndpoint, ServiceControl.Audit, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Audit.Auditing.ProcessedMessage, ServiceControl.Audit, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Audit.Auditing.FailedAuditImport, ServiceControl.Audit, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Audit.Auditing.MessagesView.MessagesViewIndex+SortAndFilterOptions, ServiceControl.Audit, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null